kernel/kernel-impl/src/main/java/org/sakaiproject/email/impl/BasicEmailService.java

```diff
@@ -1302,11 +1303,24 @@ protected void setContent(String content, List<Attachment> attachments, MimeMess
 		ArrayList<MimeBodyPart> embeddedAttachments = new ArrayList<MimeBodyPart>();
 		if (attachments != null && attachments.size() > 0)
 		{
+			int maxAttachmentSize = serverConfigurationService.getInt(MAIL_SENDFROMSAKAI_MAXSIZE, 25000000);
+			int attachmentRunningTotal = 0;
+
 			// Add attachments to messages
 			for (Attachment attachment : attachments)
 			{
 				// attach the file to the message
-				embeddedAttachments.add(createAttachmentPart(attachment));
+				MimeBodyPart mbp = createAttachmentPart(attachment);
+				int mbpSize = mbp.getSize();
+				if ( (attachmentRunningTotal + mbpSize) < maxAttachmentSize )
+				{
+					embeddedAttachments.add(mbp);
+					attachmentRunningTotal = attachmentRunningTotal + mbpSize;
+				}
+				else
+				{
+					M_log.debug("Removed attachment from mail message because it was too large: " + mbpSize);
```

Comment:
Would be helpful to log this at a higher level along with the to/from addresses so when people say an email to X arrived without an attachment there is something to find easily.